File: ai_engine/delete_broker.py

import os
from database import Database

def delete_broker_by_email(email: str):
    db = Database()
    if not db.supabase:
        print("Erro: Supabase nao inicializado.")
        return

    # 1. Busca o usuario pelo e-mail
    response = db.supabase.table("users").select("id, name").eq("email", email).execute()
    
    if not response.data:
        print(f"Usuario com e-mail {email} nao encontrado.")
        return

    user = response.data[0]
    user_id = user['id']
    user_name = user['name']

    print(f"Encontrado: {user_name} (ID: {user_id})")
    
    # 2. Deleta o usuario (O cascade no banco cuidara do resto)
    delete_response = db.supabase.table("users").delete().eq("id", user_id).execute()
    
    if delete_response.data:
        print(f"Sucesso: Corretor {user_name} ({email}) excluido com sucesso.")
    else:
        print("Erro ao excluir o corretor.")

if __name__ == "__main__":
    delete_broker_by_email("jorgemartins2k@gmail.com")

File: tmp/delete_broker.js

const postgres = require('postgres');
const dotenv = require('dotenv');
const path = require('path');

// Load .env.local from root
dotenv.config({ path: path.resolve(__dirname, '../.env.local') });

const connectionString = process.env.DATABASE_URL;

if (!connectionString) {
    console.error('ERRO: DATABASE_URL nao encontrada no .env.local');
    process.exit(1);
}

const sql = postgres(connectionString);

async function deleteBroker(email) {
    try {
        console.log(`Buscando corretor com e-mail: ${email}...`);

        const users = await sql`
            SELECT id, name FROM users WHERE email = ${email}
        `;

        if (users.length === 0) {
            console.log(`Corretor com e-mail ${email} nao encontrado.`);
            return;
        }

        const user = users[0];
        console.log(`Encontrado: ${user.name} (ID: ${user.id})`);

        console.log(`Excluindo corretor...`);
        const result = await sql`
            DELETE FROM users WHERE id = ${user.id}
        `;

        console.log(`Sucesso! Corretor ${user.name} (${email}) foi excluido.`);
        console.log('As tabelas relacionadas foram limpas via CASCADE.');

    } catch (error) {
        console.error('Erro durante a exclusao:', error);
    } finally {
        await sql.end();
    }
}

deleteBroker('jorgemartins2k@gmail.com');
